worlds/cvcotm/items.py

```diff
@@ -0,0 +1,213 @@
+                del world.options.start_inventory_from_pool.value[card]
+
+            logging.warning(f"[{world.player_name}] Too many DSS Cards in "
+                            f"Start Inventory from Pool to satisfy the Halve DSS Cards Placed option. The following "
+                            f"{len(removed_start_cards)} card(s) were removed: {removed_start_cards}")
+
+            start_cards = kept_start_cards
+
+        # Remove the starting cards from the excluded cards.
+        for card in ACTION_CARDS.union(ATTRIBUTE_CARDS):
+            if card in start_cards:
+                excluded_cards.remove(card)
+
+        # Remove a valid ice/stone action and/or attribute card if the player isn't starting with one.
+        if not has_freeze_action:
+            excluded_cards.remove(world.random.choice(FREEZE_ACTIONS))
+        if not has_freeze_attr:
+            excluded_cards.remove(world.random.choice(FREEZE_ATTRS))
+
+        # Remove 10 random cards from the exclusions.
+        excluded_items += world.random.sample(excluded_cards, 10)
+
+    # Exclude the Maiden Detonator from creation if the maidens start broken.
+    if world.options.iron_maiden_behavior == IronMaidenBehavior.option_start_broken:
+        excluded_items += [iname.ironmaidens]
+
+    # Add one of each Item to the pool that is not filler or progression skip balancing.
+    for item in cvcotm_item_info:
+        classification = cvcotm_item_info[item].default_classification
+        code = cvcotm_item_info[item].code
+
+        # Skip event Items and Items that are excluded from creation.
+        if code is None or item in excluded_items:
+            continue
+
+        # Classify the Cleansing as Useful instead of Progression if Ignore Cleansing is on.
+        if item == iname.cleansing and world.options.ignore_cleansing:
+            classification = ItemClassification.useful
+
+        # Classify the Kick Boots as Progression + Useful if Nerf Roc Wing is on.
+        if item == iname.kick_boots and world.options.nerf_roc_wing:
+            classification |= ItemClassification.useful
+
+        item_counts[classification][item] = 1
+        total_items += 1
+
+    # Add the total Last Keys if no skirmishes are required (meaning they're not forced anywhere).
+    if not world.options.required_skirmishes:
+        item_counts[ItemClassification.progression_skip_balancing][iname.last_key] = \
+            world.options.available_last_keys.value
+        total_items += world.options.available_last_keys.value
+
+    # Add filler items at random until the total Items = the total Locations.
+    while total_items < len(world.multiworld.get_unfilled_locations(world.player)):
+        filler_to_add = world.random.choice(FILLER_ITEM_NAMES)
+        if filler_to_add in item_counts[ItemClassification.filler]:
+            item_counts[ItemClassification.filler][filler_to_add] += 1
+        else:
+            item_counts[ItemClassification.filler][filler_to_add] = 1
```

Comment:
Recommend looking at `collections.Counter` here, or `collections.defaultdict(int)`. These allow you to always `+= 1`, that if the element isn't present yet, inserts it with 0 first automatically.

Not actually important though.

*low priority*

Reply:
I didn't know about collections before, so this information was very helpful, lol.